src/process/process.rs

```diff
@@ -177,58 +176,47 @@ impl Process {
 		virtual_memory: &memory::VirtualMemory,
 	) -> ProcessResult<Percent> {
 		let memory_info = self.memory_info()?;
-		let percent = ((memory_info.rss() as f64 / virtual_memory.total() as f64) * 100.0) as f32;
-
-		Ok(percent)
+		let percent = (memory_info.rss() as f64 / virtual_memory.total() as f64) * 100.0;
+		Ok(percent as f32)
 	}
 
 	pub fn memory_percent_with_type(&self, r#type: MemType) -> ProcessResult<Percent> {
 		self.sys_memory_percent_with_type(r#type)
 	}
 
-	pub fn chidren(&self) {
-		self.sys_chidren()
+	pub fn children(&self) {
+		self.sys_children()
 	}
 
 	pub fn open_files(&self) -> ProcessResult<Vec<OpenFile>> {
 		self.sys_open_files()
 	}
 
 	pub fn connections(&self) {
-		self.sys_connections()
+		todo!()
 	}
 
-	pub fn connections_with_type(&self, r#type: NetConnectionType) {
-		self.sys_connections_with_type(r#type)
+	pub fn connections_with_type(&self, r#_type: NetConnectionType) {
+		todo!()
 	}
 
 	pub fn is_running(&self) -> bool {
-		match Process::new(self.pid) {
-			Ok(p) => p == *self,
-			Err(_) => false,
-		}
+		Process::new(self.pid).map(|p| p == *self).unwrap_or(false)
 	}
 
 	/// New method, not in Python psutil.
 	pub fn is_replaced(&self) -> bool {
-		match Process::new(self.pid) {
-			Ok(p) => p != *self,
-			Err(_) => false,
-		}
+		Process::new(self.pid).map(|p| p != *self).unwrap_or(false)
 	}
 
 	/// New method, not in Python psutil.
 	pub fn replace(&mut self) -> bool {
 		match Process::new(self.pid) {
-			Ok(p) => {
-				if p == *self {
-					false
-				} else {
-					let _ = mem::replace(self, p);
-					true
-				}
+			Ok(p) if &p != self => {
+				*self = p;
```

Comment:
This line does the same as mem::replace? Otherwise this change looks good.

Reply:
`mem::replace` "remembers" the original value of the first argument (this value is returned from the `mem::replace` call), and then assigns the second argument to the first argument.

Reply:
Okay great. I think the match structure was better before though on second glance. It's a little harder to track now I think.

Reply:
I think I disagree on this point because the `match` statement seems much more compact and straight-forward to me compared to the original and it doesn't seem to suffer from the problems of the "haskellization" of the `.map.unwrap` chains of the other (reverted) modifications.